x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1247,6 +1253,36 @@ Collection<Object> createComponents(
         return components;
     }
 
+    private CustomApiKeyAuthenticator createCustomApiKeyAuthenticator(SecurityExtension.SecurityComponents extensionComponents) {
+        final SetOnce<CustomApiKeyAuthenticator> customApiKeyAuthenticatorSetOnce = new SetOnce<>();
```

Comment:
It's weird to use a `SetOnce` here rather than just a nullable `CustomApiKeyAuthenticator`
There's no threading, so I don't understand what `SetOnce` gives us.

Is there a reason?

Reply:
Good catch. I don't see a reason either.

I've pushed https://github.com/elastic/elasticsearch/pull/128440/commits/6fae14b46f66e9293e2a7bd93760073fdd54f209 which uses a `Map<String, CustomApiKeyAuthenticator>` instead. It allows adding a better error handling when more than one extension provides a `CustomApiKeyAuthenticator`.